sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -1632,16 +1611,18 @@ reduction(T *VarPtr, const T &Identity, BinaryOperation BOp) {
 /// operation used in the reduction.
 /// The identity value is not passed to this version as it is statically known.
 template <typename T, class BinaryOperation>
-detail::enable_if_t<detail::IsKnownIdentityOp<T, BinaryOperation>::value,
-                    detail::reduction_impl<T, BinaryOperation, 0, true,
-                                           access::mode::read_write>>
+std::enable_if_t<detail::IsKnownIdentityOp<T, BinaryOperation>::value,
+                 detail::reduction_impl<T, BinaryOperation, 0, true,
+                                        access::mode::read_write>>
 reduction(T *VarPtr, BinaryOperation) {
   return detail::reduction_impl<T, BinaryOperation, 0, true,
                                 access::mode::read_write>(VarPtr);
 }
 
+} // namespace ONEAPI
+
 template <typename BinaryOperation, typename AccumulatorT>
-struct has_known_identity : detail::has_known_identity_impl<
+struct has_known_identity : ONEAPI::detail::has_known_identity_impl<
```

Comment:
I don't know the answer to this, but I want to make sure that we're being consistent across different extensions and SYCL 2020 features.  Should this be available as `ONEAPI::has_known_identity` in SYCL 1.2.1 mode, and only available as `has_known_identity` in SYCL 2020 mode?

Reply:
> only available as has_known_identity in SYCL 2020 mode

John, I tried implementing what you suggested here: 9b355efa81966655860c15d8d88c0a0a393d3bcd
When you say "only in SYCL 2020 mode" what do you mean? I thought about using macro SYCL_LANGUAGE_VERSION automatically set with "-fsycl", but that now seems a bit too severe to me as in this case it is not possible to use has_known_identity without using -fsycl switch (and without explicit define of SYCL_LANGUAGE_VERSION macro).

Reply:
I was assuming that we would still allow developers to request SYCL 1.2.1 -- if that isn't the case, then I'm less worried about conditionally enabling SYCL 2020 features.

That said, I think standard C++ headers typically guard features based on the standard that included them.  I don't feel strongly about this -- @rolandschulz, do you think it matters at all?

Reply:
If we don't (partially) support multiple SYCL versions, I don't see a reason to have features guarded.